.idea/inspectionProfiles/MCreator.xml

```diff
@@ -22,6 +22,9 @@
       <option name="m_reportAllNonLibraryCalls" value="false" />
       <option name="callCheckString" value="java.io.File,.*,java.io.InputStream,read|skip|available|markSupported,java.io.Reader,read|skip|ready|markSupported,java.lang.Boolean,.*,java.lang.Byte,.*,java.lang.Character,.*,java.lang.Double,.*,java.lang.Float,.*,java.lang.Integer,.*,java.lang.Long,.*,java.lang.Math,.*,java.lang.Object,equals|hashCode|toString,java.lang.Short,.*,java.lang.StrictMath,.*,java.lang.String,.*,java.math.BigInteger,.*,java.math.BigDecimal,.*,java.net.InetAddress,.*,java.net.URI,.*,java.util.UUID,.*,java.util.regex.Matcher,pattern|toMatchResult|start|end|group|groupCount|matches|find|lookingAt|quoteReplacement|replaceAll|replaceFirst|regionStart|regionEnd|hasTransparantBounds|hasAnchoringBounds|hitEnd|requireEnd,java.util.regex.Pattern,.*" />
     </inspection_tool>
+    <inspection_tool class="Java9CollectionFactory" enabled="true" level="WEAK WARNING" enabled_by_default="true">
```

Comment:
This should be reversed

Reply:
Without those changes, I get a warning from the compiler on these lines:
https://github.com/MCreator/MCreator/blob/5ca3344d105f4b0eadf8f132113d6cba89ac1ecc/src/main/java/net/mcreator/generator/Generator.java#L490-L493
It suggests me to replace `Collections.unmodifiableList(new ArrayList<>(...))` with `List.copyOf(...)` but this factory method doesn't e.g. handle nulls properly, and I don't think there is an OKish way I can suppress it.

Reply:
We can't disable it fully like this, as warning has a sense probably. Either ignore the warning or suppress it for statement only if possible

Reply:
`IGNORE_NON_CONSTANT` stands for <kbd>Only report when content is constant</kbd> in IDEA's UI.
>Use the **Do not warn when content is non-constant** option to report only in cases when the supplied arguments are compile-time constants. This reduces the chances that the behavior changes, because it's not always possible to statically check whether original elements are unique and not `null`.

Reply:
If this can't be surpressed for statement using comment, ignore warning, do not ignore it in XML

Reply:
✔️ Fixed 😔

Reply:
* You just saw the explanation?
* Add it back?

Reply:
> You just saw the explanation?

Your explanation was:

>  but this factory method doesn't e.g. handle nulls properly, and I don't think there is an OKish way I can suppress it.

but this is not true as handling is same as with ArrayList, it will throw NPE

So I am not sure what other reason there is to create array list instead of using List.copyOf which is probably recommended by IDE because the method has JIT optimizations likely

Reply:
I removed explanation from IDEA because I'm not sure if I may post text from this program.

Reply:
It is open-source. I still can not understand why you don't want to do what is suggests as your explanation is not true

Reply:
Well in your case, both methods (redundant ArrayList or List.copyOf) handle null the same way (you can verify this yourself in JavaDoc), so ignoring this warning will not change anything.

Thus I would recommend using a more modern List.copyOf variant as the IDE suggested

Reply:
>you can verify this yourself in JavaDoc

My bad, sorry 🤦‍♂️

Reply:
Verified
>@throws NullPointerException if the specified collection is null

(from `new ArrayList<>()`)

Reply:
`List.copyOf()` does call `Objects.requireNonNull()` on each element from the array, `new ArrayList<>()` does not. Did I read the wrong docs?

Reply:
requireNonNull will throw NPE, so will ArrayList construction with null.

So there is no benefit not following advice from IDE and depressing this warning

Reply:
>requireNonNull will throw NPE, so will ArrayList construction with null. _Both will throw NPE._

`ArrayList` constructor throws exception if the array argument is null. Factory method `List.copyOf()` throws it if **either array or _any of its elements_** is null. **This** is what `IGNORE_NON_CONSTANT` option helps with.

Reply:
Oops, autocorrected supressing to depressing lol :D

Reply:
> Factory method List.copyOf() throws it if either array or any of its elements is null.

Isn't that good in fact, if a certain element will be null for list data, that template will fail to generate. Getting NPE here will, however, make sure we catch those cases in tests

Reply:
Warning seem to be gone:

![image](https://user-images.githubusercontent.com/16374228/210402560-82caa047-a475-4fae-bb37-86524b8fdca4.png)

Reply:
The XML configuration for `IGNORE_NON_CONSTANT`

Reply:
Not needed if there is no warning to be suppressed anymore I would say